### DISCUSSION TITLE
Lamp and Inverted Lamp rendering Halo option

### DISCUSSION THREAD

**Discussion by cap1o** [MrTJP/ProjectRed#1643] [Ideas]
Hello, 

I have some troubles with lamps that weren't rendering the halo on great distance.
I have build a tower with inverted lamp and at the top the halo is off and on the bottom it's on.
It's quite weird.

It could be a good thing to let the user decide if he want or not this effect on distance.
At least, shut down light when it's not visible.
Hoshiko
![2021-07-04_16 01 04](https://user-images.githubusercontent.com/29143461/124390589-94192300-dcec-11eb-9fe3-33be56ad86c9.png)

**Comment by MrTJP** [comment]
This is a vanilla mechanic. Tile entities are not rendered once you get far away. You should see a similar effect for things like chests, etc. 

This may improve in the future once I move lamp rendering outside of the TESR.

**Reply by cap1o** [reply]
I understand it's a tile entities but i want to tell you some french have already fix it with a custom mod.
But it's old and it not anymore online ^^. That's why i ask you if something is possible about that. 

https://www.youtube.com/watch?v=KqP85SDHGV0&list=PLbSdidkZzKrjTfMFPu722KYaaV8TQ_Yco

It's in french but talk about on the serie.

Thanks for your mod anyway i play with it since years.

**Reply by MrTJP** [reply]
I have a ‘solution’, which is to not use the TESR. That’s how we did it before 1.15 but it does not work the same anymore. I’ll look into it more.

**Reply by cap1o** [reply]
Thank you ^^ Let's build and make some flashing lights !
![2021-07-04_22 42 30](https://user-images.githubusercontent.com/29143461/124399962-60a3bc00-dd1f-11eb-834f-7df735e563ed.png)

**Comment by MrTJP** [comment]
These issues have been resolved for a while now. Probably when the Fabulous Lighting was implemented (https://github.com/MrTJP/ProjectRed/discussions/1775), or maybe a little before that.
